runtime/src/snapshot_utils.rs

```diff
@@ -798,6 +1213,75 @@ fn sort_snapshot_archives(snapshot_archives: &mut Vec<SnapshotArchiveInfo>) {
     snapshot_archives.sort_unstable_by(|a, b| b.slot.cmp(&a.slot));
 }
 
+/// Get a list of the incremental snapshot archives in a directory
+fn get_incremental_snapshot_archives<P>(
+    snapshot_output_dir: P,
+) -> Vec<IncrementalSnapshotArchiveInfo>
+where
+    P: AsRef<Path>,
+{
+    match fs::read_dir(&snapshot_output_dir) {
+        Err(err) => {
+            info!(
+                "Unable to read snapshot directory: err: {}, path: {}",
+                err,
+                snapshot_output_dir.as_ref().display()
+            );
+            vec![]
+        }
+        Ok(files) => files
+            .filter_map(|entry| {
+                if let Ok(entry) = entry {
+                    let path = entry.path();
+                    if path.is_file() {
+                        if let Some((base_slot, slot, hash, archive_format)) =
+                            parse_incremental_snapshot_archive_filename(
+                                path.file_name().unwrap().to_str().unwrap(),
+                            )
+                        {
+                            return Some(IncrementalSnapshotArchiveInfo {
+                                inner: SnapshotArchiveInfo {
+                                    path,
+                                    slot,
+                                    hash,
+                                    archive_format,
+                                },
+                                base_slot,
+                            });
+                        }
```

Comment:
maybe extract to `IncrementalSnapshotArchiveInfo::new_from_path(path) -> Result<Self>`?

Reply:
Done!